modules/heat_conduction/src/meshgenerators/PatchSidesetGenerator.C

```diff
@@ -188,8 +193,23 @@ PatchSidesetGenerator::generate()
 
   // partition the boundary mesh
   boundary_mesh->prepare_for_use();
-  MooseMesh::setPartitioner(*boundary_mesh, _partitioner_name, false, _pars, *this);
-  boundary_mesh->partition(_n_patches);
+  _n_boundary_mesh_elems = boundary_mesh->n_elem();
+  if (_partitioner_name == "grid")
+    partition(*boundary_mesh);
+  else
+  {
+    // the enum of this class and MooseMesh enum are different
+    // so we have to go through these intermediate steps to not
+    // mess things up with partitioner
+    std::string pstring = _partitioner_name;
+    MooseEnum penum = MooseMesh::partitioning();
+    penum = pstring;
+    MooseMesh::setPartitioner(*boundary_mesh, penum, false, _pars, *this);
+    boundary_mesh->partition(_n_patches);
```

Comment:
I don't think you need the initial copy
```suggestion
    MooseEnum penum = MooseMesh::partitioning();
    penum = _partitioner_name;
    MooseMesh::setPartitioner(*boundary_mesh, penum, false, _pars, *this);
    boundary_mesh->partition(_n_patches);
```